Add eslint config for max line length 100

bronzehedwick opened the pull request:
The `max-len` rule was not specified, and not included in
prettier/eslint defaults for some reason; tested by adding the `max-len`
rule at agreed upon 100 columns, and seeing 2 strings exceeding that
line length in dispatch.js fail.

Mark the rule as an error, but allow it to ignore long strings and URLs;
in the case of the former, we would have to change the line wrapping of
the output text if we changed the source, and that doesn't seem worth
it. URLs aren't correct if they break to a new line so those should be
excluded as well.

James-Quigley commented:
Is the 80 col rule enforced by prettier or eslint?

bronzehedwick commented:
@James-Quigley Neither, just a general guideline. If we're okay with adding a rule to either to enforce this I'm happy to do that, or if we'd rather only conform style to what's currently enforced I can decline this PR.

James-Quigley commented:
@bronzehedwick yeah I'd say lets make sure we have a rule to enforce it as well. That way it will be continually kept in check too.

azlyth commented:
I think 80 is a bit aggressive for modern screens, how do people feel about 100?

@bcosynot you have thoughts on what line length should be?

bronzehedwick commented:
I think this [reddit thread ](https://www.reddit.com/r/programming/comments/2nkntp/does_column_width_80_make_sense_in_2014/) makes a good argument for 80 columns.

In summary, the argument is that:

1. It's more readable for humans to view any text at around 60 columns, max 75, so 80 was reached to give more syntax-noise breathing room
2. Code is read in places other than an editor, such as diffs, code review, etc, that may have more constrained width
3. People still have small laptop screens, or work with multiple windows sizes (fairly) narrow

bcosynot commented:
If I am not mistaken eslint (and/or prettier) enforces a line length of 100. I am usually a fan of leaving defaults on with things like this so I don't have to think about it.

bronzehedwick commented:
Did some discovery, and it doesn't look like the `max-len` eslint rule was being applied. I revert/rebased my 80 column code, and added eslint config that adds `max-len` 100.

Added my new commit message to the description.
